File: app/frontend/ea-project/src/models/output/GetCompetitionByIdOutput.js
import BaseOutput from "./BaseOutput";
import GenderEnumerator from "../enums/Gender";
import GradeEnumerator from "../enums/Grade";

export default class GetCompetitionByIdOutput extends BaseOutput {
  constructor(id = 0, name = '', startDate = new Date(), endDate = new Date(), isActive = false, creationDate = new Date(),gender="",grade="",category="", feedbackMessages = []) {
    super(feedbackMessages);
    this.id = id;
    this.name = name;
    this.startDate = startDate;
    this.endDate = endDate;
    this.isActive = isActive;
    this.creationDate = creationDate;
    for (const key in GenderEnumerator) {
      if (key === gender) {
        this.gender = GenderEnumerator[key];
        console.log("aqui")
        break;
      }
    }
    
    for (const key in GradeEnumerator) {
      if (key === grade) {
        this.grade = GradeEnumerator[key];
        console.log("aqui")
        break;
      }
    }
    this.type=category;
    this.trials=[];
  }
}

File: app/backend/src/main/java/eaproject/output/GetCompetitionByIdOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;

public class GetCompetitionByIdOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private ArrayList<GetAllTrialsOutput.TrialProperties> trials;

    @JsonProperty
    private Integer id;

    @JsonProperty
    private String name;

    @JsonProperty
    private Timestamp startDate;

    @JsonProperty
    private Timestamp endDate;

    @JsonProperty
    private boolean isActive;

    @JsonProperty
    private Timestamp creationDate;

    public ArrayList<GetAllTrialsOutput.TrialProperties> getTrials() {
        return trials;
    }

    public void setTrials(ArrayList<GetAllTrialsOutput.TrialProperties> trials) {
        this.trials = trials;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Timestamp getStartDate() {
        return startDate;
    }

    public void setStartDate(Timestamp startDate) {
        this.startDate = startDate;
    }

    public Timestamp getEndDate() {
        return endDate;
    }

    public void setEndDate(Timestamp endDate) {
        this.endDate = endDate;
    }

    public boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(boolean isActive) {
        this.isActive = isActive;
    }

    public Timestamp getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }


}
